apiserver/facades/controller/undertaker/undertaker_test.go

```diff
@@ -173,23 +217,32 @@ func (s *undertakerSuite) TestDeadRemoveModel(c *gc.C) {
 
 	// Set model to dead
 	otherSt.model.life = state.Dying
-	err := hostedAPI.ProcessDyingModel(context.Background())
+	err = hostedAPI.ProcessDyingModel(context.Background())
 	c.Assert(err, gc.IsNil)
 
 	err = hostedAPI.RemoveModel(context.Background())
 	c.Assert(err, jc.ErrorIsNil)
 
 	c.Assert(otherSt.removed, jc.IsTrue)
-	c.Assert(s.secrets.cleanedUUID, gc.Equals, otherSt.model.uuid)
+
+	c.Assert(s.secrets.cleanedUUID, gc.Equals, modelInfo.UUID.String())
 }
 
 func (s *undertakerSuite) TestDeadRemoveModelSecretsConfigNotFound(c *gc.C) {
 	otherSt, hostedAPI, ctrl := s.setupStateAndAPI(c, false, "hostedmodel")
 	defer ctrl.Finish()
-	s.mockSecretBackendService.EXPECT().GetSecretBackendConfigForAdmin(gomock.Any(), coremodel.UUID(otherSt.model.uuid)).Return(nil, secretbackenderrors.NotFound)
+
+	s.mockModelInfoService.EXPECT().GetModelInfo(gomock.Any()).Return(coremodel.ModelInfo{
+		UUID: "9d3d3b19-2b0c-4a3f-acde-0b1645586a72",
+	}, nil).Times(2)
+
+	modelInfo, err := hostedAPI.modelInfoService.GetModelInfo(context.Background())
```

Comment:
nit: Perhaps we could refactor this context.Background() out to be reused? CMIIW but the actual implementation should use the same ctx in each api flow

Reply:
fixed.